### PR TITLE
feat(topic): add retention size flag for topic create

### LINKED ISSUE #523: Add retention size flag to create topic command
### Feature or problem description

`rhoas kafka topic create` should have a flag `--retention-size`.

### Proposed solution

`--retention-size` should have a default value -1. It represents `bytes`.
Also needs to be handled in the interactive mode.

### PR DESCRIPTION
**Pull request by rkpattnaik780**
### Description

`topic create` should have a flag for setting the retention size.

fixes #523 

### Verification Steps

1. Create topics using both normal and interactive mode.
2. Normal mode should support a `--retention-bytes` flag.
3. Interactive mode should prompt for retention size.

### Type of change

- [X] New feature (non-breaking change which adds functionality)

### Checklist

- [X] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: pkg/cmd/kafka/topic/create/create.go [hunk 1]
```diff
@@ -268,13 +271,24 @@ func runInteractivePrompt(opts *Options) (err error) {
...
+		return err
+	}
+
+	retentionBytesPrompt := &survey.Input{
+		Message: localizer.MustLocalizeFromID("kafka.topic.create.input.retentionBytes.message"),
+		Help:    localizer.MustLocalizeFromID("kafka.topic.common.input.retentionBytes.description"),
+		Default: fmt.Sprintf("%v", defaultRetentionSize),
+	}
+
+	err = survey.AskOne(retentionBytesPrompt, &opts.retentionBytes, survey.WithValidator(topicutil.ValidateMessageRetentionPeriod))
```
Comment (craicoverflow) [inline pkg/cmd/kafka/topic/create/create.go:291]: This is using the message retention period validator.

File: pkg/cmd/kafka/topic/create/create.go [hunk 2]
```diff
@@ -116,6 +118,7 @@ func NewCreateTopicCommand(f *factory.Factory) *cobra.Command {
 	}))
 	cmd.Flags().Int32Var(&opts.partitions, "partitions", 1, localizer.MustLocalizeFromID("kafka.topic.common.input.partitions.description"))
 	cmd.Flags().IntVar(&opts.retentionMs, "retention-ms", defaultRetentionPeriodMS, localizer.MustLocalizeFromID("kafka.topic.common.input.retentionMs.description"))
+	cmd.Flags().IntVar(&opts.retentionBytes, "retention-bytes", defaultRetentionSize, localizer.MustLocalizeFromID("kafka.topic.common.input.retentionBytes.description"))
```
Comment (craicoverflow) [inline pkg/cmd/kafka/topic/create/create.go:121]: The flag value is not being validated anywhere, I can submit invalid values

File: pkg/kafka/topic/validators.go [hunk 3]
```diff
@@ -154,6 +154,38 @@ func ValidateMessageRetentionPeriod(v interface{}) error {
 	return nil
 }
 
+// ValidateMessageRetentionPeriod validates the value (bytes) of the retention size
```
Comment (craicoverflow) [inline pkg/kafka/topic/validators.go:157]:
```suggestion
// ValidateMessageRetentionSize validates the value (bytes) of the retention size
```

**Review by craicoverflow — CHANGES_REQUESTED** [review]
```
❯ rhoas kafka topic list
  NAME (6)         PARTITIONS   RETENTION TIME (MS)   RETENTION SIZE (BYTES)  
 ---------------- ------------ --------------------- ------------------------        
  topic-4                 100   604800000             -2                      
  topic-5                   1   604800000             -2                    
```

I should not be able to enter `-2` as a value.

This flag will also be needed in the update command.

**Comment by rkpattnaik780** [comment]
> ```
> ❯ rhoas kafka topic list
>   NAME (6)         PARTITIONS   RETENTION TIME (MS)   RETENTION SIZE (BYTES)  
>  ---------------- ------------ --------------------- ------------------------        
>   topic-4                 100   604800000             -2                      
>   topic-5                   1   604800000             -2                    
> ```
> 
> I should not be able to enter `-2` as a value.
> 
> This flag will also be needed in the update command.

was thinking to add this to update command in a separate PR.

**Review by craicoverflow — APPROVED** [review]

**Pull request merged**